services/catarse.js/legacy/src/c/input-card.tsx

```diff
@@ -0,0 +1,35 @@
+import m from 'mithril'
+import { Event } from '../entities';
+
+export type InputCardProps = {
+    cardClass: string
+    onclick(event : Event<MouseEvent>): void
+    label: string
+    label_hint: string
+    children: m.ChildArrayOrPrimitive
+    belowChildren: any
+}
+
+export default class InputCard {
+    view({attrs, children} : m.Vnode<InputCardProps>) {
+        const {
+            cardClass = '.u-marginbottom-30.card.card-terciary',
+            onclick = Function,
+            label,
+            label_hint,
+            children: attrsChildren,
+            belowChildren,
+        } = attrs;
+
+        return m(cardClass, { onclick }, [
```

Comment:
**Dúvida:**
Percebi que esse arquivo foi recriado, mudando de .js pra .tsx mas ele continua com a sintaxe de criação de componentes antiga do mithiril. Tem algum motivo por trás disso? Qual os fatos que definem se vamos criar usando a sintaxe nova ou a antiga?

Reply:
@stephannv esse foi para deixar preparado quando for fazer algo mais estrutural na edição do projeto, como por exemplo, na home de projeto. 

O critério é trocar todas as vezes, exceto quando esse componente é usado em mais partes do que o escopo da tarefa deveria cubrir, mas merece um "revisit" no futuo. Isso não está documentado mesmo, mas é garantir o mínimo de refatoração sem ultrapassar muito o escopo da tarefa/projeto. Eu troquei a extensão, mas não a interface, para iniciar os tipos ali para ele.